static_analysis/Dockerfile

```diff
@@ -0,0 +1,86 @@
+#######
+# cppcheck build
+######
+
+FROM alpine:latest AS cppcheck_build
+
+# Version 1.90
+ARG cppcheck_commit=077e652
+
+RUN \
+    apk add git ninja cmake g++ pcre-dev && \
+    \
+    git clone https://github.com/danmar/cppcheck.git
+
+WORKDIR /cppcheck
+
+RUN git checkout ${cppcheck_commit} && mkdir build
+
+WORKDIR /cppcheck/build
+RUN CXXFLAGS="--static" cmake .. -DCMAKE_BUILD_TYPE=Release -DHAVE_RULES=ON -DUSE_MATCHCOMPILER=ON -G Ninja && \
+    cmake --build .
+
+
+#######
+# Clang-Format build adapted from https://github.com/Unibeautify/docker-beautifiers
+#######
+# MIT License
+# 
+# Copyright (c) 2017 Unibeautify
+# 
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+# 
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+# 
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+FROM alpine:latest as clang-format-build
+
+# Build dependencies
+RUN apk update && apk add git build-base ninja cmake python3
+
+ARG LLVM_TAG="release/9.x"
+
+# Download and setup
+WORKDIR /build
+RUN git clone --branch ${LLVM_TAG} --depth 1 https://github.com/llvm/llvm-project.git
+WORKDIR /build/llvm-project
+RUN mv clang llvm/tools
+RUN mv libcxx llvm/projects
+
+# Build
+WORKDIR llvm/build
+RUN cmake -GNinja -DLLVM_BUILD_STATIC=ON -DLLVM_ENABLE_LIBCXX=ON ..
```

Comment:
Given that `jenkins/jnlp-slave` is Debian based is there a reason we are not just using the pre-built binaries for Kitware?

Reply:
Debian stretch's version of `clang-format` is too old, I can't seem to find Kitware's PPA
Could you post a link to it as I'd rather drop this if possible